File: daily/2025/2211_Count_Collisions_on_a_Road.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    int countCollisions(string directions) {
        int n = directions.length();
        int ans = directions.length();
        for (int i = 0; i < n; i++){
            if (directions[i] == 'L') ans--;
            else break;
        }
        for (int i = n - 1; i >= 0; i--){
            if (directions[i] == 'R') ans--;
            else break;
        }
        for (char c : directions){
            if (c == 'S') ans--;
        }
        return ans;
    }
};

int main(){
    string directions = "RLRSLL";
    // string directions = "LLRR";
    Solution S;
    cout << S.countCollisions(directions) << endl;
    return 0;
}

File: daily_python/2211_Count_Collisions_on_a_Road.py
import os
import sys

class Solution:
    def countCollisions(self, directions: str) -> int:
        n = len(directions)
        l, r = 0, n - 1
        while l < n and directions[l] == 'L':
            l += 1
        while r >= 0 and directions[r] == 'R':
            r -= 1
        ans = 0
        for i in range(l, r + 1):
            if directions[i] != 'S':
                ans += 1
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            directions = lines[0].split('=')[1].strip()[1:-2]
            dataset.append(directions)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for directions in dataset:
        results.append(solution.countCollisions(directions))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')